MediaStore/app/src/main/java/com/android/samples/mediastore/MainActivity.kt

```diff
@@ -113,6 +135,15 @@ class MainActivity : AppCompatActivity() {
         }
     }
 
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (resultCode == Activity.RESULT_OK && requestCode == DELETE_PERMISSION_REQUEST) {
+            viewModel.pendingDeleteImage?.let { image ->
+                viewModel.deleteImage(image)
+            }
```

Comment:
Move this logic to a ViewModel fun: `deletePendingImage()` or something

Reply:
Done.